sm/plugin_sm/src/logged_command.rs

```diff
@@ -0,0 +1,185 @@
+use std::process::{Output, Stdio};
+use tokio::fs::File;
+use tokio::io::{AsyncWriteExt, BufWriter};
+use tokio::process::Command;
+
+pub struct LoggedCommand {
+    command_line: String,
+    command: Command,
+}
+
+impl std::fmt::Display for LoggedCommand {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.command_line.fmt(f)
+    }
+}
+
+impl LoggedCommand {
+    pub fn new(program: &str) -> LoggedCommand {
+        let command_line = program.to_string();
+        let mut command = Command::new(program);
+        command
+            .current_dir("/tmp")
+            .stdin(Stdio::null())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped());
+
+        LoggedCommand {
+            command_line,
+            command,
+        }
+    }
+
+    pub fn arg(&mut self, arg: &str) -> &mut LoggedCommand {
+        self.command_line.push_str(&format!(" {:?}", arg));
```

Comment:
I'm not sure this requires `{:?}`, I think you can just do:
```suggestion
        self.command_line.push_str(arg);
```

Reply:
The arguments are displayed as debug, to be properly quoted and distinguished from each other.

=> I added a comment